library/src/main/java/com/novoda/downloadmanager/DownloadsBatchPersistence.java

```diff
@@ -44,15 +44,14 @@ void persistAsync(DownloadBatchTitle downloadBatchTitle,
                         notificationSeen
                 );
                 downloadsPersistence.persistBatch(batchPersisted);
-
-                for (DownloadFile downloadFile : downloadFiles) {
-                    downloadFile.persistSync();
-                }
-
                 downloadsPersistence.transactionSuccess();
             } finally {
                 downloadsPersistence.endTransaction();
             }
+
+            for (DownloadFile downloadFile : downloadFiles) {
```

Comment:
Why was this moved?

Reply:
now the `downloadFile.persistSync` has his own transaction operations

Reply:
I was wondering if we should move this transaction logic to the `DownloadsPersistence` rather than having it everywhere else.

Reply:
Downside would be that any new `DownloadPersistence` would have to pay attention to implement the transactions too 🤔